compiler/optimizer/VPHandlers.cpp

```diff
@@ -4187,8 +4187,12 @@ TR::Node *constrainNewArray(OMR::ValuePropagation *vp, TR::Node *node)
       dumpOptDetails(vp->comp(), "size node has no known constraint for newarray %p\n", sizeNode);
       //node->setAllocationCanBeRemoved(false)
       }
-   else
+   else if (sizeConstraint->getLowInt() >= 0 && sizeConstraint->getLowInt() <= maxSize)
```

Comment:
Looks like this should check `getHighInt() <= maxSize` (rather than `getLowInt()`), here and in `constrainANewArray()` below

Reply:
Sigh!  Yes.

Reply:
Fixed this in commit f69a2e82d40235de90493904a9342490fdcd9c94